macrosynergy/download/dataquery.py

```diff
@@ -247,36 +277,38 @@ def __init__(
             "aud": self.__dq_api_resource_id,
         }
 
-        # For debugging purposes save last request response.
-        self.status_code: Optional[int] = None
-        self.last_response: Optional[str] = None
-        self.last_url: Optional[str] = None
-        self.proxy: Optional[dict] = proxy
-
-    def _active_token(self) -> bool:
-        """Confirms if the token being used has not expired."""
-        created: datetime = self._stored_token["created_at"]
-        expires: int = self._stored_token["expires_in"]
+    def _valid_token(self) -> bool:
+        """
+        Method to check if the stored token is valid.
 
-        return (datetime.now() - created).total_seconds() / 60 >= (expires - 1)
+        :return <bool>: True if the token is valid, False otherwise.
+        """
+        if self._stored_token is None:
+            return False
 
-    def _valid_token(self) -> bool:
-        """Confirms if the credentials passed correspond to a valid token."""
-        return not (self._stored_token is None or self._active_token())
+        created: datetime.datetime = self._stored_token["created_at"]
+        expires: int = self._stored_token["expires_in"]
+        is_active = (datetime.now() - created).total_seconds() / 60 >= (expires - 1)
+        return is_active
 
     def _get_token(self) -> str:
```

Comment:
Refactor to use `from oauthlib.oauth2 import BackendApplicationClient` and `from requests_oauthlib import OAuth2Session`.

Reply:
Saving this particular method for a later update and resolving this comment.